src/components/structures/RightPanel.js

```diff
@@ -187,19 +186,22 @@ export default class RightPanel extends React.Component {
             panel = <GroupMemberList groupId={this.props.groupId} key={this.props.groupId} />;
         } else if (this.state.phase === RIGHT_PANEL_PHASES.GroupRoomList) {
             panel = <GroupRoomList groupId={this.props.groupId} key={this.props.groupId} />;
-        } else if (this.state.phase === RIGHT_PANEL_PHASES.RoomMemberInfo) {
+        } else if (this.state.phase === RIGHT_PANEL_PHASES.RoomMemberInfo ||
+            this.state.phase === RIGHT_PANEL_PHASES.EncryptionPanel) {
             if (SettingsStore.isFeatureEnabled("feature_cross_signing")) {
                 const onClose = () => {
                     dis.dispatch({
                         action: "view_user",
-                        member: null,
+                        member: this.state.phase === RIGHT_PANEL_PHASES.EncryptionPanel ? this.state.member : null,
```

Comment:
Does it also fix https://github.com/vector-im/riot-web/issues/11989 then?

Reply:
it doesn't, that phase doesn't get entered upon the first verification trigger with the right panel closed, this just changes it so that if you're in verification phase the close button goes back to UserInfo but when in base user info you go back to Member List